node_modules/oae-core/resetpassword/js/resetpassword.js

```diff
@@ -0,0 +1,159 @@
+/*!
+ * Copyright 2014 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+define(['jquery','underscore','oae.core'], function($, _, oae) {
+
+    return function(uid, showSettings) {
+		
+		// The widget container
+		var $rootel = $('#' + uid);
+    
+    /**
+     * Verify whether or not the entered username already exists as a login id on the current tenant.
+     * The ajax request in this function executes synchronously to allow a username to be checked for
+     * existence before the user is created.
+     *
+     * @param  {String}     username                The username we want to check the existence of
+     * @param  {Function}   callback                Standard callback function.
+     * @param  {Boolean}    callback.available      Whether or not the username is available
+     */
+    var isUserNameAvailable = function(username, callback) {
```

Comment:
It looks like we do the same or very similar check in several different places - could consolidate all into a single utility instead of repeating code.